File: kotlinCRUD/src/main/kotlin/com/example/kotlinCRUD/Employee.java
package com.example.kotlinCRUD;

import javax.persistence.GeneratedValue;
import javax.persistence.Id;

public class Employee {
    private @Id @GeneratedValue Long id;
    private String name;
    private String role;

    Employee(){}

    Employee(String name,String role){
        this.name = name;
        this.role = role;
    }

    public Long getId(){
        return this.id;
    }

    public String getName(){
        return this.name;
    }
    public  String getRole(){
        return this.role;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setRole(String role) {
        this.role = role;
    }

    @Override
    public boolean equ
}

File: SpringBootAppJPA/src/main/kotlin/com/example/springbootappjpa/Employee.kt
package com.example.springbootappjpa

import java.util.*
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.Id



@Entity
internal class Employee {
    @Id
    @GeneratedValue
    var id: Long? = null
    var name: String? = null
    var role: String? = null

    constructor() {}
    constructor(name: String?, role: String?) {
        this.name = name
        this.role = role
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o !is Employee) return false
        val employee = o
        return (id == employee.id && name == employee.name
                && role == employee.role)
    }

    override fun hashCode(): Int {
        return Objects.hash(id, name, role)
    }

    override fun toString(): String {
        return "Employee{" + "id=" + id + ", name='" + name + '\'' + ", role='" + role + '\'' + '}'
    }
}